src/kernel.hpp

```diff
@@ -95,22 +91,23 @@ using cvk_kernel_holder = refcounted_holder<cvk_kernel>;
 
 struct cvk_kernel_argument_values {
 
-    cvk_kernel_argument_values(cvk_kernel* kernel, uint32_t num_resources)
-        : m_kernel(kernel), m_pod_buffer(nullptr), m_owns_resources(false),
-          m_kernel_resources(num_resources),
-          m_local_args_size(m_kernel->num_args(), 0) {}
+    cvk_kernel_argument_values(cvk_entry_point* entry_point)
+        : m_entry_point(entry_point), m_is_enqueued(false),
+          m_args(m_entry_point->args()), m_pod_arg(nullptr),
+          m_kernel_resources(m_entry_point->num_resources()),
+          m_local_args_size(m_entry_point->args().size(), 0) {}
 
     cvk_kernel_argument_values(const cvk_kernel_argument_values& other)
-        : m_kernel(other.m_kernel), m_pod_buffer(nullptr),
-          m_owns_resources(false), m_kernel_resources(other.m_kernel_resources),
+        : m_entry_point(other.m_entry_point), m_is_enqueued(false),
+          m_args(m_entry_point->args()), m_pod_arg(nullptr),
+          m_kernel_resources(other.m_kernel_resources),
           m_local_args_size(other.m_local_args_size) {}
 
-    ~cvk_kernel_argument_values() { release_resources(); }
+    ~cvk_kernel_argument_values() {}
```

Comment:
The destructor should probably free the descriptor sets.

Reply:
Whoops, don't know how I missed that :-/

Fixed.

Reply:
Thanks. I almost missed it as well. Neither the validation layers nor valgrind caught it.